packages/experimental/Text/src/Text.tsx

```diff
@@ -39,56 +42,45 @@ export type TextProps<TBase = ITextProps> = TBase &
     disabled?: boolean;
   };
 
-/**
- * These are the tokens which are also present in props. If specified in props this will override the values
- * from tokens.
- */
-const tokensThatAreAlsoProps: (keyof TextTokens)[] = ['variant', 'color'];
+const useTextTokens = buildUseTokens<TextTokens>(
+  (t) => ({
+    variant: 'secondaryStandard',
+    color: t.colors.bodyText,
+    disabled: {
+      color: t.colors.disabledText,
+    },
+  }),
+  textName,
+);
 
-/** Type to use for compose */
-interface TextType {
-  props: TextProps;
-  slotProps: { root: ITextProps };
-  tokens: TextTokens;
-}
+export const Text = compressible<TextProps, TextTokens>((props: TextProps, useTokens: UseTokens<TextTokens>) => {
+  // split out color and variant from props
+  const { color, variant, style, disabled, ...rest } = props;
+  const theme = useFluentTheme();
+  // get the tokens from the theme
+  let [tokens, cache] = useTokens(theme);
 
-export const Text = compose<TextType>({
-  displayName: textName,
-  /** Settings for the use-styling hook */
-  tokens: [
-    (t) => ({
-      variant: 'secondaryStandard',
-      color: t.colors.bodyText,
-      disabled: {
-        color: t.colors.disabledText,
-      },
+  // apply states like disabled if specified in props
+  [tokens, cache] = applyTokenLayers(tokens, ['disabled'], cache, (state) => props[state]);
+  // override variant or color from props
+  [tokens, cache] = patchTokens(tokens, cache, { color, variant });
+
+  // now build the text style from tokens that can be shared between different Text instances
+  const [tokenStyle] = cache(
+    () => ({
+      margin: 0,
+      color: tokens.color,
+      ...fontStyles.from(tokens, theme),
     }),
-    textName,
-  ],
-  states: ['disabled'],
-  tokensThatAreAlsoProps,
-  slotProps: {
-    root: buildProps<ITextProps, TextTokens>(
-      (tokens: TextTokens, theme: Theme) => ({
-        style: {
-          margin: 0,
-          color: tokens.color,
-          ...fontStyles.from(tokens, theme),
-        },
-      }),
-      ['color', ...fontStyles.keys],
-    ),
-  },
-  /** Settings for the useSlots that will be passed on */
-  slots: { root: RNText },
-  filters: { root: filterTextProps },
-  /** render function for the component */
-  render: (props: TextProps, useSlots: UseSlots<TextType>) => {
-    // stage one, execute any hooks, styling lookups to build the styled slot
-    const Root = useSlots(props).root;
-    // return a function used to complete the render
-    return (rest: TextProps, children: React.ReactNode) => <Root {...mergeProps(props, rest)}>{children}</Root>;
-  },
-});
+    [],
+  );
+
+  // return a continuation function that allows this text to be compressed
+  return (extra: TextProps, children: React.ReactNode) => {
+    const mergedProps = { ...rest, ...extra, style: mergeStyles(tokenStyle, props.style, extra.style) };
+    return <RNText {...mergedProps}>{children}</RNText>;
+  };
```

Comment:
I hadn't thought of how to phrase this in your previous PR, but do we have a description (in documentation or elsewhere) of what it means to be `compressed`? As in, if someone knew to the repo (who knows not of using slots to optimize down the number of renders) comes and looks at this component, would they know what this means?

Reply:
We do in the use-slot README, but likely need to agree on consistent terminology here. I think I've referred to it in the past as a staged component, but that is phrasing things in terms of the implementation details rather than the goal. Maybe I should change the word compressed to compressible here?

Reply:
I think compressible makes sense. In the use-slot README, I saw references to stagedComponent but not compressible. Am I looking at the wrong README? I'm hoping that the right README is easy enough to find by someone ctrl+clicking on `compressible` in VS Code (since that's how I've been going about it). Or also a comment above the function definition when you do a mouse hover over

Reply:
I linked Saad our side conversation about the naming here, maybe we should've renamed stagedComponent to compressible like you originally suggested? Can you still use compressible without customize? (If so, does that basically make it the same as stagedComponent?) or did we make the useTokens param required?

Reply:
The reason for my comment here is less about the name, and making the name's meaning discoverable to someone who is looking at this code for the first time. The name feels appropriate to how I understand it, but I imagine if this was my first time looking at the code I wouldn't have semantically understood what compressible meant, that it was related to stagedComponent, or that the README in use-slots is the right place to learn that info.

Reply:
I would +1 this. I've been thinking that it would be good to do a pass and comment all of these functions with JSDoc styling so that when you're typing with something that has Intellisense, the function's purpose is easy to see (similar to what happens when you use things from RN). Personally wouldn't find this blocking, but I think it would make it easier to look at the framework and work with it if there's documentation accessible from the IDE itself.

Reply:
That makes sense. I would propose the overall pattern use the words compress or compressible in discussions, then I would likely rename stagedComponent to stagedRender as the implementation detail. I would likely create an alias as well (const stagedComponent = stagedRender) to reduce the risk of breaks. I would do that in a separate PR I think.

Reply:
Thanks. I'm starting to feel that some of these things should be split from the PR like the `useFluentTheme` hook and documentation/intellisense documentation updates.